File: ios/Sources/VoiceRecorder/Core/ErrorCodes.swift
import Foundation

/// Canonical error codes returned by the plugin.
struct ErrorCodes {
    static let missingPermission = "MISSING_PERMISSION"
    static let alreadyRecording = "ALREADY_RECORDING"
    static let microphoneBeingUsed = "MICROPHONE_BEING_USED"
    static let deviceCannotVoiceRecord = "DEVICE_CANNOT_VOICE_RECORD"
    static let failedToRecord = "FAILED_TO_RECORD"
    static let emptyRecording = "EMPTY_RECORDING"
    static let recordingHasNotStarted = "RECORDING_HAS_NOT_STARTED"
    static let failedToFetchRecording = "FAILED_TO_FETCH_RECORDING"
    static let failedToMergeRecording = "FAILED_TO_MERGE_RECORDING"
    static let notSupportedOsVersion = "NOT_SUPPORTED_OS_VERSION"
    static let couldNotQueryPermissionStatus = "COULD_NOT_QUERY_PERMISSION_STATUS"
}

File: android/src/main/java/app/independo/capacitorvoicerecorder/core/ErrorCodes.java
package app.independo.capacitorvoicerecorder.core;

/** Canonical error codes returned by the plugin. */
public final class ErrorCodes {

    public static final String MISSING_PERMISSION = "MISSING_PERMISSION";
    public static final String ALREADY_RECORDING = "ALREADY_RECORDING";
    public static final String MICROPHONE_BEING_USED = "MICROPHONE_BEING_USED";
    public static final String DEVICE_CANNOT_VOICE_RECORD = "DEVICE_CANNOT_VOICE_RECORD";
    public static final String FAILED_TO_RECORD = "FAILED_TO_RECORD";
    public static final String EMPTY_RECORDING = "EMPTY_RECORDING";
    public static final String RECORDING_HAS_NOT_STARTED = "RECORDING_HAS_NOT_STARTED";
    public static final String FAILED_TO_FETCH_RECORDING = "FAILED_TO_FETCH_RECORDING";
    public static final String FAILED_TO_MERGE_RECORDING = "FAILED_TO_MERGE_RECORDING";
    public static final String NOT_SUPPORTED_OS_VERSION = "NOT_SUPPORTED_OS_VERSION";
    public static final String COULD_NOT_QUERY_PERMISSION_STATUS = "COULD_NOT_QUERY_PERMISSION_STATUS";

    private ErrorCodes() {}
}
